python/ray/util/deprecation.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import functools
+
+logger = logging.getLogger(__name__)
+
+# A constant to use for any configuration that should be deprecated
+# (to check, whether this config has actually been assigned a proper value or
+# not).
+DEPRECATED_VALUE = -1
+
+
+def deprecation_message(msg, error=None):
```

Comment:
Adding these seems a little bit heavy-weight to me for something like deprecating APIs (which shouldn't happen often). Can we just raise `DeprecationWarning` instead?

Reply:
@edoakes Updated, please take a look, thanks.